### PR TITLE
PIMS-1602 Error Refresh

### PR DESCRIPTION
**Pull request by dbarkowsky**
## 🎯 Summary

  
[PIMS-1602](https://citz-imb.atlassian.net/browse/PIMS-1602)

## Changes
- Moves ConfigContext up one component, now around the error boundary
- Adds some logic to ErrorFallback.tsx that records the count of errors previously and redirects to the root page if it has already happened.

Idea is that the root page will hopefully have no errors, so users won't be stuck in the repeating error loop if they are trying to reload a page that is causing the error in the first place. 

## Testing
I used this replacement for UsersManagement to trigger errors:
<details><summary>UsersManagement</summary>
<p>

```typescript
const UsersManagement = () => {
  const navigate = useNavigate();
  // Getting data from API
  const usersApi = usePimsApi();
  const { data, refreshData, isLoading, error } = useDataLoader(usersApi.users.getAllUsers);

  return (
    <>
      {error.map((e) => (
        <>hi</>
      ))}
      <UsersTable
        rowClickHandler={(params) => {
          // Checking length of selection so it only navigates if user isn't trying to select something
          const selection = window.getSelection().toString();
          if (!selection.length) {
            navigate(`/users/${params.id}`);
          }
        }}
        data={data}
        refreshData={refreshData}
        isLoading={isLoading}
        error={error}
      />
    </>
  );
};
``` 
</p>
</details> 

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

**Comment by dbarkowsky** [comment]
Converted back to draft because I noticed an issue with the reporting.

**Review by GrahamS-Quartech — APPROVED** [review]
Working well. I think using session storage is okay for this, can't really leverage any react hooks with the way this works. The only other way I think we could approach this is using the URL params (ie `?errorRedirect=1`) but not sure if there's any particular advantage over this.
